### PR TITLE
Add the operator Reference API plus a typo fix

### PR DESCRIPTION
**Pull request by tmjd**
## Description

- Adding the operator reference API
- Fix typo in OpenShift curl commands

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit f419f7e246612198675b1b9de17b467a47cdf3af

https://deploy-preview-3157--calico.netlify.com

File: reference/installation/api.html [hunk 1]
```diff
@@ -0,0 +1,739 @@
+---
+title: Installation reference
+description: Installation API reference
+layout: docwithnav
+---
+
+The Kubernetes resources below configure {{site.prodname}} installation. Each resource is responsible for installing and configuring a different subsystem of {{site.prodname}}, and gets created as part of [custom-resources.yaml]({{site.baseurl}}/manifests/custom-resources.yaml) during installation. Most options can be modified on a running cluster using `kubectl`.
```
Comment (lmm) [inline reference/installation/api.html]: The `custom-resources.yaml` link is broken

File: reference/installation/api.html [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+<code>kind</code><br>
+string
+</td>
+<td><code>Installation</code></td>
+</tr>
+<tr>
+<td>
+<code>metadata</code><br>
+<em>
+<a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#objectmeta-v1-meta">
```
Comment (lmm) [inline reference/installation/api.html]: A nit but it'd be nice if these links pointed to a recent k8s version.

File: reference/installation/README.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+# Generating API reference docs
+
+The api.html doc in this directory is generated using https://github.com/caseydavenport/gen-crd-api-reference-docs.
```
Comment (lmm) [inline reference/installation/README.md]: Thanks for documenting this! But I'm wondering whether if location is the place for this information. It's useful for us but maybe not docs users. WDYT about `CONTRIBUTING_DOCS.md`?

Comment (tmjd) [reply reference/installation/README.md]: This doc does not show up in the website so I don't users would come across it. I'd like to keep it here so it is obvious how to generate this doc.

**Pull request merged**